File: app/src/main/java/com/x7/steambadger/MainActivity.java
package com.x7.steambadger;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;

import com.j256.ormlite.dao.Dao;
import com.j256.ormlite.dao.DaoManager;
import com.x7.steambadger.activity.LoginActivity;
import com.x7.steambadger.application.Config;
import com.x7.steambadger.database.DbOpenHelper;
import com.x7.steambadger.database.model.Player;
import com.x7.steambadger.fragment.ProfileFragment;
import com.x7.steambadger.fragment.menu.MenuFragment;

import java.util.List;


public class MainActivity extends AppCompatActivity {

    private Player player;

    private DrawerLayout mDrawerLayout;
    private ActionBarDrawerToggle mDrawerToggle;

    private Fragment mContent = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (Config.getInstance().getSteamId().isEmpty()) {
            Intent intent = new Intent(this, LoginActivity.class);
            startActivity(intent);
            return;
        }

        //DbOpenHelper.getInstance().dropTables(null, DbOpenHelper.getCon());
        DbOpenHelper.getInstance().createTables(null, DbOpenHelper.getCon());

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setHomeButtonEnabled(true);

        mDrawerToggle = new ActionBarDrawerToggle(this, mDrawerLayout, R.string.menu, R.string.menu);

        mDrawerLayout.addDrawerListener(mDrawerToggle);

        if (savedInstanceState != null) {
            mContent = getSupportFragmentManager().getFragment(savedInstanceState, "mContent");
        }

        getLoggedUser();

        if (mContent == null) {
            Bundle bundle = new Bundle();
            bundle.putSerializable("player", player);
            mContent = new ProfileFragment();
            mContent.setArguments(bundle);
        }

        startView();
    }

    public Player getPlayer() {
        return player;
    }

    private void startView() {
        // Menu
        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.left_drawer, new MenuFragment())
                .commit();

        // Content
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.content_frame, mContent)
                .commit();
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

        if (mDrawerToggle != null) {
            mDrawerToggle.syncState();
        }
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);

        if (mDrawerToggle != null) {
            mDrawerToggle.onConfigurationChanged(newConfig);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.toogleMenu();
                break;
            default:
                return super.onOptionsItemSelected(item);
        }

        return true;
    }

    public void toogleMenu() {
        if (mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
            mDrawerLayout.closeDrawer(GravityCompat.START);
        } else {
            mDrawerLayout.openDrawer(GravityCompat.START);
        }
    }

    @Override
    public void onBackPressed() {
        if (mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
            mDrawerLayout.closeDrawer(GravityCompat.START);
            return;
        }

        if (navigateBack()) {
            return;
        }

        this.moveTaskToBack(true);
    }

    public boolean navigateBack() {
        if (getSupportFragmentManager().popBackStackImmediate()) {
            mContent = getSupportFragmentManager().findFragmentById(R.id.content_frame);
            return true;
        }

        return false;
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        if (mContent != null) {
            super.onSaveInstanceState(outState);

            getSupportFragmentManager().putFragment(outState, "mContent", getSupportFragmentManager().findFragmentById(R.id.content_frame));
        }
    }

    public void logoff() {
        Config.getInstance().setSteamId(null);
        Config.getInstance().setCustomUrl(null);
        loginActivity();
    }

    public void loginActivity() {
        Intent intent = new Intent(this, LoginActivity.class);
        startActivity(intent);

        finish();
    }

    public void switchContent(Fragment fragment) {
        switchContent(fragment, false);
    }

    public void switchContent(Fragment fragment, boolean dontStore) {
        mDrawerLayout.closeDrawer(GravityCompat.START);

        Fragment currentFragment = getSupportFragmentManager().findFragmentById(R.id.content_frame);

        if (currentFragment != null && currentFragment.getClass() == fragment.getClass()) {
            return;
        }

        mContent = fragment;

        FragmentTransaction tx = getSupportFragmentManager().beginTransaction();

        tx.replace(R.id.content_frame, fragment);

        if (!dontStore) {
            tx.addToBackStack(null);
        }

        tx.commit();
    }

    private void getLoggedUser() {
        try {
            String steamId = Config.getInstance().getSteamId();

            Dao<Player, Long> playerDao = DaoManager.createDao(DbOpenHelper.getCon(), Player.class);
            List<Player> playerResult = playerDao.queryForEq("steamId", steamId);

            if (playerResult.isEmpty()) {
                player = new Player();
                player.setSteamId(steamId);
                playerDao.create(player);

                playerDao.refresh(player);
            } else {
                player = playerResult.get(0);
            }
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }

}

File: app/src/main/java/com/dougmaitelli/steambadger/MainActivity.kt
package com.dougmaitelli.steambadger

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import android.view.MenuItem

import com.j256.ormlite.dao.DaoManager
import com.dougmaitelli.steambadger.activity.LoginActivity
import com.dougmaitelli.steambadger.application.Config
import com.dougmaitelli.steambadger.application.SteamBadgeR
import com.dougmaitelli.steambadger.database.DatabaseHelper
import com.dougmaitelli.steambadger.database.model.Player
import com.dougmaitelli.steambadger.fragment.ProfileFragment
import com.dougmaitelli.steambadger.fragment.menu.MenuFragment
import com.google.firebase.analytics.FirebaseAnalytics

class MainActivity : AppCompatActivity() {

    var player: Player? = null

    private lateinit var firebaseAnalytics: FirebaseAnalytics

    private var mDrawerLayout: DrawerLayout? = null
    private var mDrawerToggle: ActionBarDrawerToggle? = null

    private var mMenu: Fragment? = null
    private var mContent: Fragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        firebaseAnalytics = FirebaseAnalytics.getInstance(this)

        if (Config.getInstance(this).steamId.isNullOrEmpty()) {
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
            return
        }

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        mDrawerLayout = findViewById(R.id.drawer_layout)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeButtonEnabled(true)

        mDrawerToggle = ActionBarDrawerToggle(this, mDrawerLayout, R.string.menu, R.string.menu)

        mDrawerLayout!!.addDrawerListener(mDrawerToggle!!)

        if (savedInstanceState != null) {
            mMenu = supportFragmentManager.getFragment(savedInstanceState, "mMenu")
            mContent = supportFragmentManager.getFragment(savedInstanceState, "mContent")
        }

        getLoggedUser()

        if (mMenu == null) {
            mMenu = MenuFragment()
        }

        if (mContent == null) {
            val bundle = Bundle()
            bundle.putSerializable("player", player)
            mContent = ProfileFragment()
            mContent!!.arguments = bundle
        }

        startView()
    }

    private fun startView() {
        // Menu
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.left_drawer, mMenu!!)
                .commit()

        // Content
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.content_frame, mContent!!)
                .commit()
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)

        if (mDrawerToggle != null) {
            mDrawerToggle!!.syncState()
        }
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)

        if (mDrawerToggle != null) {
            mDrawerToggle!!.onConfigurationChanged(newConfig)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> this.toogleMenu()
            else -> return super.onOptionsItemSelected(item)
        }

        return true
    }

    private fun toogleMenu() {
        if (mDrawerToggle == null) {
            return
        }

        if (mDrawerLayout!!.isDrawerOpen(GravityCompat.START)) {
            mDrawerLayout!!.closeDrawer(GravityCompat.START)
        } else {
            mDrawerLayout!!.openDrawer(GravityCompat.START)
        }
    }

    override fun onBackPressed() {
        if (mDrawerToggle == null) {
            return
        }

        if (mDrawerLayout!!.isDrawerOpen(GravityCompat.START)) {
            mDrawerLayout!!.closeDrawer(GravityCompat.START)
            return
        }

        if (navigateBack()) {
            return
        }

        this.moveTaskToBack(true)
    }

    private fun navigateBack(): Boolean {
        if (supportFragmentManager.popBackStackImmediate()) {
            mContent = supportFragmentManager.findFragmentById(R.id.content_frame)
            return true
        }

        return false
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        if (mMenu != null) {
            supportFragmentManager.putFragment(outState, "mMenu", supportFragmentManager.findFragmentById(R.id.left_drawer)!!)
        }

        if (mContent != null) {
            supportFragmentManager.putFragment(outState, "mContent", supportFragmentManager.findFragmentById(R.id.content_frame)!!)
        }
    }

    fun logoff() {
        val customUrl = Config.getInstance(this).customUrl

        Config.getInstance(this).setSteamId(null)
        Config.getInstance(this).setCustomUrl(null)

        val bundle = Bundle()
        bundle.putString(SteamBadgeR.Param.CUSTOM_URL, customUrl)
        firebaseAnalytics.logEvent(SteamBadgeR.Event.LOGOUT, bundle)

        loginActivity()
    }

    private fun loginActivity() {
        val intent = Intent(this, LoginActivity::class.java)
        startActivity(intent)

        finish()
    }

    @JvmOverloads
    fun switchContent(fragment: Fragment, dontStore: Boolean = false) {
        mDrawerLayout!!.closeDrawer(GravityCompat.START)

        val currentFragment = supportFragmentManager.findFragmentById(R.id.content_frame)

        if (currentFragment != null && currentFragment.javaClass === fragment.javaClass) {
            return
        }

        mContent = fragment

        val tx = supportFragmentManager.beginTransaction()

        tx.replace(R.id.content_frame, fragment)

        if (!dontStore) {
            tx.addToBackStack(null)
        }

        tx.commit()
    }

    private fun getLoggedUser() {
        try {
            val steamId = Config.getInstance(this).steamId

            val playerDao = DaoManager.createDao(DatabaseHelper.connectionSource, Player::class.java)
            val playerResult = playerDao.queryForEq("steamId", steamId)

            if (playerResult.isEmpty()) {
                player = Player()
                player!!.steamId = steamId
                playerDao.create(player)

                playerDao.refresh(player)
            } else {
                player = playerResult[0]
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }

    }

}
